client/my-sites/backup/backup-storage-space/backup-storage-space-upsell/index.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useCallback, useEffect } from 'react';
+import { Button } from '@wordpress/components';
+import { sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+import { useTranslate, translate as translateType } from 'i18n-calypso';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type BackupStorageSpaceUpsellOptions =
+	| 'no_upsell'
+	| 'first_upsell'
+	| 'second_upsell'
+	| 'out_of_storage';
+
+const getStatusText = (
+	upsellOption: BackupStorageSpaceUpsellOptions,
+	storageLimit: number,
+	translate: typeof translateType
+) => {
+	switch ( upsellOption ) {
+		case 'first_upsell':
+			// TODO: calculate storage time, account for GB, and translate once API data is available.
+			return sprintf( 'You will reach your %1$sGB storage limit in %2$s days', storageLimit, 3 );
+		case 'second_upsell':
+			return translate( 'You’re running out of storage space.' );
+		case 'out_of_storage':
+			return translate( 'You ran out of storage space.' );
+		case 'no_upsell':
+		default:
+			return '';
+	}
+};
+
+type Props = {
+	href: string;
+	storageLimit: number;
+	upsellOption: BackupStorageSpaceUpsellOptions;
+	usedStorage: number;
+};
+
+export const BackupStorageSpaceUpsell: FunctionComponent< Props > = ( {
+	href,
+	storageLimit,
+	upsellOption,
+	usedStorage,
+} ) => {
+	const translate = useTranslate();
+
+	useEffect( () => {
+		recordTracksEvent( 'calypso_jetpack_backup_storage_upsell_display', {
+			type: upsellOption,
+			usedStorage,
+		} );
+	}, [ upsellOption, usedStorage ] );
```

Comment:
We'll need to wrap all calls to `recordTracksEvent` in `dispatch` to make sure they fire correctly:

```suggestion
	useEffect( () => {
		dispatch( recordTracksEvent( 'calypso_jetpack_backup_storage_upsell_display', {
			type: upsellOption,
			usedStorage,
		} ) );
	}, [ dispatch, upsellOption, usedStorage ] );
```

Reply:
Good call, updated.